KsApi/models/ErrorEnvelope.swift

```diff
@@ -120,71 +129,70 @@ public struct ErrorEnvelope {
-      <*> json <| "http_code"
-      <*> json <|? "exception"
-      <*> json <|? "facebook_user"
-      <*> .success(nil)
-
-    // ...but sometimes we make requests to the www server and JSON errors come back in a different envelope
-    let nonStandardErrorEnvelope = {
-      curry(ErrorEnvelope.init)
-        <^> concatSuccesses([
-          json <|| ["data", "errors", "amount"],
-          json <|| ["data", "errors", "backer_reward"]
-        ])
-        <*> .success(ErrorEnvelope.KsrCode.UnknownCode)
-        <*> json <| "status"
-        <*> .success(nil)
-        <*> .success(nil)
-        <*> .success(nil)
-    }
+extension ErrorEnvelope: Swift.Decodable {
+  enum CodingKeys: String, CodingKey {
+    case errorMessages = "error_messages"
+    case ksrCode = "ksr_code"
+    case httpCode = "http_code"
+    case status
+    case data
+    case exception
+    case facebookUser = "facebook_user"
+  }
 
-    return standardErrorEnvelope <|> nonStandardErrorEnvelope()
+  public init(from decoder: Decoder) throws {
+    // sometimes we make requests to the www server and JSON errors come back in a different envelope
+    let values = try decoder.container(keyedBy: CodingKeys.self)
+    if values.contains(.httpCode) {
+      self.errorMessages = try values.decode([String].self, forKey: .errorMessages)
+      self.ksrCode = try values.decodeIfPresent(KsrCode.self, forKey: .ksrCode)
+      self.httpCode = try values.decode(Int.self, forKey: .httpCode)
+      self.exception = try values.decode(Exception.self, forKey: .exception)
+      self.facebookUser = try values.decodeIfPresent(FacebookUser.self, forKey: .facebookUser)
+      self.graphError = nil
+      self.data = nil
+    } else {
+      self.data = try values.decodeIfPresent(ErrorEnvelope.AltErrorMessage.self, forKey: .data)
+      self.errorMessages = self.data?.errors.details ?? []
+      self.ksrCode = ErrorEnvelope.KsrCode.UnknownCode
+      self.httpCode = try values.decode(Int.self, forKey: .status)
+      self.exception = nil
+      self.facebookUser = nil
+      self.graphError = nil
+    }
   }
 }
 
-extension ErrorEnvelope.Exception: Argo.Decodable {
-  public static func decode(_ json: JSON) -> Decoded<ErrorEnvelope.Exception> {
-    return curry(ErrorEnvelope.Exception.init)
-      <^> json <||? "backtrace"
-      <*> json <|? "message"
+extension ErrorEnvelope.AltErrorMessage: Swift.Decodable {}
+
+extension ErrorEnvelope.AltErrorMessage.AltErrorDetails: Swift.Decodable {
+  enum CodingKeys: String, CodingKey {
+    case amount
+    case backerReward = "backer_reward"
   }
-}
 
-extension ErrorEnvelope.KsrCode: Argo.Decodable {
-  public static func decode(_ j: JSON) -> Decoded<ErrorEnvelope.KsrCode> {
-    switch j {
-    case let .string(s):
-      return pure(ErrorEnvelope.KsrCode(rawValue: s) ?? ErrorEnvelope.KsrCode.UnknownCode)
-    default:
-      return .typeMismatch(expected: "ErrorEnvelope.KsrCode", actual: j)
+  public init(from decoder: Decoder) throws {
```

Comment:
Also wondering if nil-coalescing would work here? Not sure how  `try` affects the use of that operator.